File: app/src/main/java/net/tlalka/android/fiszki/activities/BasePageActivity.java
package net.tlalka.android.fiszki.activities;

import android.view.Menu;
import android.view.MenuItem;
import net.tlalka.android.fiszki.R;
import net.tlalka.android.fiszki.elements.SetupElement;
import net.tlalka.android.fiszki.models.DbHelper;
import net.tlalka.android.fiszki.models.DbManager;
import net.tlalka.android.fiszki.utils.ValidUtils;

public abstract class BasePageActivity extends AbstractActivity {

    private DbHelper dbHelper;

    protected DbHelper getDbHelper() {
        if (ValidUtils.isNull(this.dbHelper)) {
            this.dbHelper = DbManager.getHelper(this);
        }
        return this.dbHelper;
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (ValidUtils.isNotNull(dbHelper)) {
            DbManager.releaseHelper();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        return this.createMenu(R.menu.page_menu, menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        return SetupElement.triggerAction(this, item.getItemId()) || super.onOptionsItemSelected(item);
    }
}

File: app/src/main/kotlin/net/tlalka/fiszki/view/activities/BasePageActivity.kt
package net.tlalka.fiszki.view.activities

import android.view.Menu
import android.view.MenuItem
import net.tlalka.fiszki.R
import net.tlalka.fiszki.view.elements.SetupElement

abstract class BasePageActivity : AbstractActivity() {

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        return createMenu(R.menu.page_menu, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return SetupElement.triggerAction(this, item.itemId) || super.onOptionsItemSelected(item)
    }
}
